packages/flutter/lib/src/widgets/scroll_activity.dart

```diff
@@ -241,10 +260,16 @@ class ScrollDragController implements Drag {
   @override
   void update(DragUpdateDetails details) {
     assert(details.primaryDelta != null);
+    final bool lastEventWasStationary = _lastDetails is DragUpdateDetails
+        && _lastDetails.primaryDelta == 0.0;
     _lastDetails = details;
     double offset = details.primaryDelta;
-    if (offset == 0.0)
+    if (offset == 0.0) {
+      // No axis motion over 2 consecutive frames breaks previous momentum.
```

Comment:
I was trying to add a timestamp to Drag*Details initially but it seems like not all places where Drag*Details are instantiated originated from a PointerEvent which had an original timestamp. Letting it be nullable or sometimes synthesized sometimes from the engine felt a bit inconsistent. 

Should I create some sort of clock interface in src/foundation or something to let it be mockable? I looked a bit but didn't find previous examples of recording timestamps directly in the framework.